components/BasicAuthLogin.tsx

```diff
@@ -0,0 +1,90 @@
+'use client';
+
+import { useState } from 'react';
+import Link from 'next/link';
+import toast from 'react-hot-toast';
+// import { z } from 'zod';
+
+import { credentialsSignIn } from '@/lib/actions';
+import { userSchema } from '@/lib/validations/userSchema';
+
+import Input from '@/components/shared/ui/Input';
+import Button from '@/components/shared/ui/Button';
+
+const BasicAuthLogin = () => {
+  const [data, setData] = useState({
+    email: '',
+    password: '',
+  });
+  // const [zodErrors, setZodErrors] = useState({});
+
+  const handleSubmit = async () => {
+    try {
+      const partialUserSchema = userSchema.partial();
+      partialUserSchema.parse(data);
+    } catch (error) {
+      console.log('zodErrors', error);
+      // if (error instanceof z.ZodError) {
+      //   setZodErrors(error.flatten());
+      // }
+      return;
+    }
+    try {
+      const { email, password } = data;
+      await credentialsSignIn({ email, password });
+    } catch (error) {
+      if (error) {
+        toast.error('Invalid user');
+      }
+    }
```

Comment:
```
import { ZodError } from 'zod'
```

```suggestion
    try {
      const partialUserSchema = userSchema.partial();
      partialUserSchema.parse(data);
      const { email, password } = data;
      await credentialsSignIn({
        email,
        password
      });
    } catch (error) {
      if (error instanceof ZodError) {
        console.log("zodErrors", error);
        // these errors come from zod
        // show the user the messages in the form
      } else {
        // these errors come from invalid signIn
        // display your toast
        toast.error("nooo");
      }
      return;
    }
```

We can consolidate our error catching by using `instance of ZodError` to figure out where the error came from.


Example of the code working:

https://github.com/DDVVPP/GitNote/assets/4108484/6d431b83-0205-40fd-bf4e-6e1b7f733169